test/login_external_test.rb

```diff
@@ -42,18 +43,44 @@ def request_session
     end
 
     describe "token is correct" do
-      let(:cookie_data){{ "foo" => "bar" }}
-      let(:token){ JWT.encode(cookie_data, shared_key, "HS256") }
+      let(:jwt_data){{ "foo" => "bar" }}
+      let(:token){ JWT.encode(jwt_data, shared_key, "HS256") }
+
+      describe "there is no active oauth authorization request" do
+        it "writes cookies and redirects to dashboard" do
+          any_instance_of(Identity::LoginExternal) do |finalize|
+            mock(finalize).write_authentication_to_cookie(jwt_data)
+          end
+
+          get "/login/external?token=#{token}"
+
+          assert_equal 302, last_response.status
+          assert_equal Identity::Config.dashboard_url, last_response.headers["Location"]
+        end
+      end
 
-      it "writes cookies and redirects to dashboard" do
-        any_instance_of(Identity::LoginExternal) do |finalize|
-          mock(finalize).write_authentication_to_cookie(cookie_data)
+      describe "the user has an active oauth authorization request" do
+        before do
+          stub_heroku_api
         end
 
-        get "/login/external?token=#{token}"
+        let(:jwt_data) do
+          { access_token: { token: 'abcd', expires_in: 9000 },
+            session: { id: '456' },
+            user: {id: '123'},
```

Comment:
Space inside { missing.<br>Space inside } missing.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.